### PR TITLE
feat: support executing nightly tests on CI

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
It instruments the CI to be able to modify the tags to be run in the CI test execution, adding or excluding the `@nightly` tag from Godog. To achieve that, we have defined a global environment variable for the tag to be added/excluded, NIGHTLY_TAG, and a Jenkins input parameter to control the nightly tags, NIGHTLY_SCENARIOS, which takes `false` as default value.

Finally, the nightly job for Fleet will trigger the main job setting the new parameter with `true` value, although it's also possible to run thee nightly tests when running the main job manually, be it a PR or a branch.

## Why is it important?
We want Fleet's nightly build to run the tests that are marked as `@nightly`.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests for the CLI, and they are passing locally
- [ ] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Related issues

- Relates #652

## Use cases

```gherkin
Scenario Outline: Triggering the tests for a <buildType> excluding the nightly tests
When the E2E tests are run for a "<buildType>" in the "<repo>" repository
   And the NIGHTLY_SCENARIOS check is not checked
Then the nightly tests are not executed
Examples:
| repo        | buildType |
| beats       | PR        |
| beats       | branch    |
| e2e-testing | PR        |
| e2e-testing | branch    |

Scenario Outline: Triggering the tests for a <buildType> including the nightly tests
When the E2E tests are run for a "<buildType>" in the "<repo>" repository
   And the NIGHTLY_SCENARIOS check is checked
Then the nightly tests are executed
Examples:
| repo        | buildType |
| beats       | PR        |
| beats       | branch    |
| e2e-testing | PR        |
| e2e-testing | branch    |

Scenario: Triggering the tests for a nightly build including the nightly tests
When the E2E tests are run for a nightly build in the e2e-testing repository
Then the nightly tests are executed
   And the NIGHTLY_SCENARIOS check is checked
```

## Follow-ups
We still have to replace the `@skip` tag from the Upgrade tests with `@nightly`, but we are going to do that in another R, as we are testing its behaviour before enabling it back.

### CODE REVIEW

**Review by kuisathaverat — APPROVED** [review]

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -312,6 +315,16 @@ def generateFunctionalTestStep(Map args = [:]){
...
   def stackVersion = env."${sneakCaseSuite}_STACK_VERSION"
   def tags = args.get('tags')
+
+  // We will decide whether to include the nightly tests in the execution at CI time, only.
+  // On the other hand, the developers can use the TAGS environment variable locally.
+  // Finally, we positively know that tags are not empty, so we can use AND operator.
+  if ("${NIGHTLY_SCENARIOS}" == "true") {
+    tags += " && ${NIGHTLY_TAG}"
+  } else {
+    tags += " && ~${NIGHTLY_TAG}"
```
Comment (mdelapenya) [inline .ci/Jenkinsfile:325]: This is incorrect. I'm designing it in another, completer manner

**Comment by mdelapenya** [comment]
jenkins run the tests please

**Comment by mdelapenya** [comment]
jenkins run the tests please

**Review by cachedout — APPROVED** [review]

**Comment by mdelapenya** [comment]
jenkins run the tests please

**Comment by mdelapenya** [comment]
Same 3 errors as in master. Merging

**Pull request merged**